test/connect/js2go.js

```diff
@@ -8,46 +8,84 @@ const expect = chai.expect
 
 const spawnDaemons = require('../utils/spawnDaemons')
 
-describe('connect', () => {
-  let daemons
+const beforeConnect = (ctx, keyType) => {
+  ctx.timeout(20 * 1000)
 
-  // Start Daemons
-  before(async function () {
-    this.timeout(20 * 1000)
+  return spawnDaemons(2, [{type: 'js', keyType}, {type: 'go', keyType}])
```

Comment:
If we change how `start` is called we can leverage the third param of `spawnDaemons` to pass `{ id: keyPath }`

Reply:
Also, right now these two daemons are using the same key, which **should** fail. We should give each daemon in the test a separate key.